targets/TARGET_STM/hal_tick_overrides.c

```diff
@@ -40,16 +42,23 @@ static uint16_t prev_tick_remainder;
 
 HAL_StatusTypeDef HAL_InitTick(uint32_t TickPriority)
 {
+    if (initialized) {
```

Comment:
thanks for your contribution
Shouldn't HAL_InitTick content also be under critical section if this can happen that HAL_InitTick is called from several threads ?

Reply:
I don't think that is needed since HAL_InitTick is called before main() is called and no user threads are running yet.